api/_utils.js

```
@@ -214,6 +214,37 @@ const getCachedTokenPrice = async (l1Token) => {
   );
 };
 
+/**
+ * Uses axios to fetch a cache-enabled rest endpoint to find if a route is enabled
+ * @param {string} originChainId The chain ID of the 'from' chain
+ * @param {string} destinationChainId The chain ID of the 'to' chain
+ * @param {string} token The token address at the origin chain
+ * @returns {Promise<{routeEnabled: boolean; errorMessage?: string}>} An object which indicates if the route is enabled and a corresponding error if it is not
+ */
+const isRouteEnabledFromCache = async (
+  originChainId,
+  destinationChainId,
+  token
+) => {
+  try {
+    console.log(
+      `INFO(isRouteEnabledFromCache): Resolving enabled-route from ${resolveVercelEndpoint()}/api/enabled-routes`
+    );
+    return (
+      await axios(`${resolveVercelEndpoint()}/api/enabled-routes`, {
+        params: { originChainId, destinationChainId, token },
+      })
+    ).data;
+  } catch (e) {
+    if (e instanceof axios.AxiosError) {
+      return e.response.data;
```

Why do we want to return the data if there was an error?

---

Hmm but this is a util function. If you swallow this error here and return something, how would the code outside know that this is an error?

---

I'm not sure why we want to do this, which seems more complicated than just rethrowing the error from axios? The outer code would catch that specific error and resurface as 500? If axios errors out, it definitely should be a 500 from our end.